### DISCUSSION TITLE
Why even in the era of TF2.x does ART not like eager execution enabled? Why stick to TF1.x style code?

### DISCUSSION THREAD

**Discussion by swapnilsayansaha** [Trusted-AI/adversarial-robustness-toolbox#1906] [Q&A] [answered]
```classifier = KerasClassifier(model=model, clip_values=(0, 1), use_logits=False)```

This line assumes you have eager execution disabled. However, for custom models, disabling eager execution leads to a lot of error. For example: https://github.com/tensorflow/tensorflow/issues/39908 

Why does ART still rely on out-of-date TF1.x style code?

**Comment by beat-buesser** [comment, marked as answer]
Hi @swapnilsayansaha ART does not rely on out-of-date TF1.x style code as you call it. For the era of TF2.x, also as you call it, and because ART likes eager execution we have created `art.estimators.classification.TensorFlowV2Classifier.` but kept `KerasClassifier` for backward-compatibility.

**Reply by christian-westbrook** [reply]
Hey @beat-buesser! This context is really helpful, and should probably be made more apparent in the documentation, especially if there's no plan to update KerasClassifier to work with TF2. As a beginner, it would have been nice to know that I should use TF2 for a Keras model even though a KerasClassifier class exists.

**Reply by beat-buesser** [reply]
Hi @christian-westbrook Good point, where would be a good place to provide this information?

**Comment by swapnilsayansaha** [comment]
Thanks. That function provided a lot of help!
